### PR TITLE
change is_override logic

### PR DESCRIPTION
**Pull request by Stoob**
if (!$dao->is_override was prior code.  IS NOT NULL not functional equivalent

Overview
----------------------------------------
_A brief description of the pull request. Try to keep it non-technical._

Before
----------------------------------------
_The current status. Please provide screenshots or gifs ([LICEcap](http://www.cockos.com/licecap/), [SilentCast](https://github.com/colinkeenan/silentcast)) where appropriate._

After
----------------------------------------
_What has been changed. Please provide screenshots or gifs ([LICEcap](http://www.cockos.com/licecap/), [SilentCast](https://github.com/colinkeenan/silentcast)) where appropriate._

Technical Details
----------------------------------------
_If the PR introduces noteworthy technical changes, please describe them here. Provide code snippets if necessary_

Comments
----------------------------------------
_Anything else you would like the reviewer to note_

### CODE REVIEW

**Comment by Stoob** [comment]
https://lab.civicrm.org/dev/core/issues/881

**Comment by seamuslee001** [comment]
paging @andrew-cormick-dockery @johntwyman RTYI

File: CRM/Member/BAO/Membership.php [hunk 1]
```diff
@@ -2272,7 +2272,7 @@ public static function updateAllMembershipStatus() {
       self::processOverriddenUntilDateMembership($dao1);
     }
 
-    $query = $baseQuery . " AND civicrm_membership.is_override IS NULL
+    $query = $baseQuery . " AND civicrm_membership.is_override != 1
```
Comment (andrew-cormick-dockery) [inline CRM/Member/BAO/Membership.php]: This won't match `is_override IS NULL`.  You need:

`AND (civicrm_membership.is_override IS NULL OR civicrm_membership.is_override == 0)`

Comment (Stoob) [reply CRM/Member/BAO/Membership.php]: such correct.  indeed i was backwards

**Comment by seamuslee001** [comment]
Jenkins re test this please

**Comment by pradpnayak** [comment]
This patch works for me

Test Case: 

Had membership with status over-ride set to 0 in 4.6, upgraded CiviCRM to 5.12

Without patch:
Ran update membership status schedule job, membership status did get updated with is_override = 0, but did got changed for is_override IS NULL

Without patch:
Ran update membership status schedule job, membership status got updated with is_override = 0 and is_override IS NULL

GOOD TO MERGE!

**Comment by eileenmcnaughton** [comment]
@Stoob is this your first merged PR?

**Comment by Stoob** [comment]
Hehe, no I don't think so @eileenmcnaughton but it's been a couple years since I did a PR that wasn't documentation related.

**Comment by eileenmcnaughton** [comment]
well good on you!

**Pull request merged**